platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesReplayer.java

```diff
@@ -170,6 +178,10 @@ public NoInput replayPces(@NonNull final IOIterator<PlatformEvent> eventIterator
                 // until it catches up before we can continue.
                 waitUntilHealthy();
 
+                if (config.limitReplayFrequency() && !rateLimiter.requestAndTrigger()) {
+                    continue;
+                }
```

Comment:
Minor: this creates a tight loop, potentially consuming more CPU than it could've. It would be nice to have a new API in the RateLimiter class that would wait/sleep until a token is available.

Reply:
I think it would be worth using such a feature if it existed, but probably not worth implementing for this use case. The slight extra CPU usage isn't a big concern at this point of startup, with the bottleneck being a single transaction handling thread.

Reply:
Sounds good.